### PR TITLE
Basic error handling and logging

### LINKED ISSUE #28: App crashes or does not respond on error
## Expected Behavior
When the app encounters an error with lighthouse or the URL entered, it should provide an informative error response.

## Actual Behavior
The app either becomes unresponsive or crashes without any indication that something happened.

## Steps to Reproduce the Problem

1. Choose a URL that breaks Lighthouse
1. Run the tool with the URL.

### PR DESCRIPTION
**Pull request by nenuadrian**
Fixes #28

> It's a good idea to open an issue first for discussion.

- [x] Tests pass
- [x] Appropriate changes to README are included in PR

### CODE REVIEW

**Comment by nenuadrian** [comment]
Screenshot 2023-01-24 at 16 55 33
Screenshot 2023-01-24 at 16 55 48

File: public/index.html [hunk 1]
```diff
@@ -67,29 +70,47 @@ <h1>3rd Party Analysis!</h1>
...
+            error.innerText = '';
+            error.style.display = 'none';
+
+            const table = document.getElementById('results');
+            table.style.display = 'none';
             var xhr = new XMLHttpRequest();
             xhr.onreadystatechange = function () {
-              if (this.readyState == 4 && this.status == 200) {
-                var result = JSON.parse(this.responseText);
+              if (this.readyState == 4) {
```
Comment (adamread) [inline public/index.html:91]: Please use the identity operator ([jsstyle](https://google.github.io/styleguide/jsguide.html#features-equality-checks))

This is also for the check for status === 200 below.

File: public/index.html [hunk 2]
```diff
@@ -43,19 +46,19 @@ <h1>3rd Party Analysis!</h1>
...
+          const table = document.getElementById('results');
+          table.style.display = 'block';
           result.forEach((result, index) => {
             const row = table.insertRow(1 + index);
-            row.classList.add("result");
+            row.classList.add('result');
             const url = row.insertCell(0);
             if (result.blockedURL.length > 70) {
-              url.innerText = result.blockedURL.substring(1, 70) + "...";
+              url.innerText = result.blockedURL.substring(1, 70) + '...';
```
Comment (adamread) [inline public/index.html:59]: I think this branch needs to be rebased on main before merging to fix this.

File: public/index.html [hunk 3]
```diff
@@ -67,29 +70,47 @@ <h1>3rd Party Analysis!</h1>
...
+              '/test/' +
+                encodeURIComponent(document.getElementById('url').value),
               true
             );
-            xhr.setRequestHeader("Content-Type", "application/json");
-            xhr.send(JSON.stringify());
+            xhr.setRequestHeader('Content-Type', 'application/json');
+            xhr.send();
           } catch (e) {
             throw new Error(e.message);
```
Comment (adamread) [inline public/index.html:115]: It seems like an error message should also be shown here.

**Review by adamread — APPROVED** [review]
Mostly LGTM. There's a == in the template and a place where I think an error message could be shown. It would nice if you fixed them before merging, please.

**Review by adamread — APPROVED** [review]
Fixed the merge with main myself, but the other comments stand.

**Pull request merged**